### PR TITLE
[SPARK-21208][R] Adds setLocalProperty and getLocalProperty in R

### PR DESCRIPTION
**Pull request by HyukjinKwon**
## What changes were proposed in this pull request?

This PR adds `setLocalProperty` and `getLocalProperty`in R.

```R
> df <- createDataFrame(iris)
> setLocalProperty("spark.job.description", "Hello world!")
> count(df)
> setLocalProperty("spark.job.description", "Hi !!")
> count(df)
```

```R
> print(getLocalProperty("spark.job.description"))
NULL
> setLocalProperty("spark.job.description", "Hello world!")
> print(getLocalProperty("spark.job.description"))
[1] "Hello world!"
> setLocalProperty("spark.job.description", "Hi !!")
> print(getLocalProperty("spark.job.description"))
[1] "Hi !!"
```

## How was this patch tested?

Manually tested and a test in `R/pkg/tests/fulltests/test_context.R`.

### CODE REVIEW

File: R/pkg/R/sparkR.R [hunk 1]
```diff
@@ -564,6 +564,23 @@ setJobDescription <- function(value) {
   invisible(callJMethod(sc, "setJobDescription", value))
 }
 
+#' Set a local property that affects jobs submitted from this thread, such as the
+#' Spark fair scheduler pool.
```
Comment (HyukjinKwon) [inline R/pkg/R/sparkR.R]: I tried to make a better description for a while but then just ended up with copying Python's: https://github.com/apache/spark/blob/209b9361ac8a4410ff797cff1115e1888e2f7e66/python/pyspark/context.py#L934-L939

**Comment by HyukjinKwon** [comment]
cc @felixcheung, could you take a look please?

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by felixcheung** [comment]
LG, I’d as.character on both the key and value to make sure it doesn’t fail with signature mismatch.

Also should we also have getLocalProperty, you think?

**Comment by HyukjinKwon** [comment]
Yea, let me update it soon.

**Comment by felixcheung** [comment]
getLocalProperty Should not be returning invisible, I think.

Also can you check if we need to fail for key == NA?

**Comment by HyukjinKwon** [comment]
Yup, they should be handled too. Let me update.

File: R/pkg/R/sparkR.R [hunk 2]
```diff
@@ -560,10 +560,55 @@ cancelJobGroup <- function(sc, groupId) {
...
+#' @examples
+#'\dontrun{
+#' setLocalProperty("spark.scheduler.pool", "poolA")
+#'}
+#' @note setLocalProperty since 2.3.0
+setLocalProperty <- function(key, value) {
+  if (is.null(key) || is.na(key)) {
+    stop("key should not be NULL or NA.")
+  }
+  if (!is.null(value)) {
```
Comment (HyukjinKwon) [inline R/pkg/R/sparkR.R]: Just to be clear, `is.na` case seems fine with passing it with `as.character`. I added the test.

**Comment by felixcheung** [comment]
Great LGTM

**Review by felixcheung — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

Thank you @felixcheung.

**Pull request closed without merging**